modules/tensor_mechanics/src/materials/GeneralizedKelvinVoigtBase.C

```diff
@@ -0,0 +1,138 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "GeneralizedKelvinVoigtBase.h"
+
+template <>
+InputParameters
+validParams<GeneralizedKelvinVoigtBase>()
+{
+  InputParameters params = validParams<LinearViscoelasticityBase>();
+  return params;
+}
+
+GeneralizedKelvinVoigtBase::GeneralizedKelvinVoigtBase(const InputParameters & parameters)
+  : LinearViscoelasticityBase(parameters)
+{
+}
+
+void
+GeneralizedKelvinVoigtBase::computeQpApparentElasticityTensors()
+{
+  _instantaneous_elasticity_tensor[_qp] = _first_elasticity_tensor[_qp];
+  _apparent_elasticity_tensor[_qp] = _first_elasticity_tensor[_qp].invSymm();
+
+  for (unsigned int i = 0; i < _springs_elasticity_tensors[_qp].size(); ++i)
+  {
+    double theta_i = computeTheta(_dt, _dashpot_viscosities[_qp][i]);
+    double gamma = _dashpot_viscosities[_qp][i] / (_dt * theta_i);
+    _apparent_elasticity_tensor[_qp] +=
+        _springs_elasticity_tensors[_qp][i].invSymm() / (1. + gamma);
```

Comment:
That is not straight forward (you are saying you effectively want to lag the elasticity tensor, right). Material properties are computed on-demand as they are needed. They are not stored (unless you make them stateful).